packages/notification-services-controller/src/NotificationServicesController/services/feature-announcements.ts

```diff
@@ -135,10 +135,14 @@ const fetchFeatureAnnouncementNotifications = async (
 export async function getFeatureAnnouncementNotifications(
   env: Env,
 ): Promise<INotification[]> {
-  const rawNotifications = await fetchFeatureAnnouncementNotifications(env);
-  const notifications = rawNotifications.map((notification) =>
-    processFeatureAnnouncement(notification),
-  );
+  if (env && env.accessToken && env.spaceId && env.platform) {
```

Comment:
🔕 Nit

```suggestion
  if (env?.accessToken && env?.spaceId && env?.platform) {
```

Or even maybe 

```
if (!env?.accessToken || !env?.spaceId || !env?.platform) return []
```

Reply:
YEEES, good callout

Reply:
I usually like early returns, but with multiple ORs it can be a little confusing. I'll leave this as is, but I would like to use ZOD or a proper shape validation.